server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/CreateAction.java

```diff
@@ -107,17 +108,17 @@ public void define(WebService.NewController controller) {
 
   @Override
   public void handle(Request request, Response response) throws Exception {
-    qProfileWsSupport.checkQProfileAdminPermission();
-
-    CreateRequest createRequest = toRequest(request);
     try (DbSession dbSession = dbClient.openSession(false)) {
-      writeProtobuf(doHandle(dbSession, createRequest, request), request, response);
+      OrganizationDto organization = qProfileWsSupport.getOrganizationByKey(dbSession, request.param(PARAM_ORGANIZATION));
+      qProfileWsSupport.checkQProfileAdminPermission(organization);
```

Comment:
`checkPermission` takes either a OrganizationDto or an organization **uuid**. In both cases we need to query the database, because the request only contains the organization key. Do you see any other solution?